Add OpenBSD virtualization facts.

eest opened the pull request:
##### ISSUE TYPE
- Feature Pull Request
##### COMPONENT NAME

facts
##### ANSIBLE VERSION

```
ansible 2.2.0 (openbsd_vm_facts df66dc4acc) last updated 2016/08/24 18:15:04 (GMT +200)
  lib/ansible/modules/core: (detached HEAD 2f26352e49) last updated 2016/08/24 18:02:24 (GMT +200)
  lib/ansible/modules/extras: (detached HEAD 65eba72ee6) last updated 2016/08/24 18:02:24 (GMT +200)
  config file = 
  configured module search path = Default w/o overrides
```
##### SUMMARY

Add OpenBSD virtualization facts.

<!-- Paste verbatim command output below, e.g. before and after your change -->

Before:

```
$ ansible -i hosts -u root 192.168.1.60 -m setup | grep ansible_virtual 
        "ansible_virtualization_role": "", 
        "ansible_virtualization_type": "", 

```

After:

```
$ ansible -i hosts -u root 192.168.1.60 -m setup | grep ansible_virtual
        "ansible_virtualization_role": "guest", 
        "ansible_virtualization_type": "virtualbox", 
```

Patch written by @jasperla.

Tested by various people on:
- virtualbox
- vmware esx(i) + fusion
- kvm (smartos + plain linux + a random cloud provider)

This patch is already present in the OpenBSD port of ansible.

eest committed:
Add OpenBSD virtualization facts.

Patch written by @jasperla.

Tested by various people on:
- virtualbox
- vmware esx(i) + fusion
- kvm (smartos + plain linux + a random cloud provider)

This patch is already present in the OpenBSD port of ansible.

bcoca commented on lib/ansible/module_utils/facts.py:
```diff
@@ -3169,8 +3169,47 @@ def populate(self):
+          self.facts['virtualization_type'] = 'RHEV'
+          self.facts['virtualization_role'] = 'guest'
+          return
+
```
use an else here and you can have only 1 return at the end (even that is not needed)

eest committed:
Rework diff to get rid of extra returns.

Requested by @bcoca.

While here, use four-space indentations of all code blocks.

eest committed:
Set facts even if no match is found.

Discussed with @bcoca.

bcoca commented on lib/ansible/module_utils/facts.py:
```diff
@@ -3169,8 +3169,42 @@ def populate(self):
     def get_virtual_facts(self):
-        self.facts['virtualization_type'] = ''
-        self.facts['virtualization_role'] = ''
+        rc, out, err = self.module.run_command("/usr/sbin/sysctl -n hw.product")
```
I would use module.get_bin_path function instead of hardcoding path (future proof), same below

eest committed:
Find sysctl via get_bin_path().

Requested by @bcoca.

eest commented:
@bcoca: Updated.

eest committed:
Fail if we do not find a sysctl binary.

bcoca commented on lib/ansible/module_utils/facts.py:
```diff
@@ -3169,8 +3169,44 @@ def populate(self):
     def get_virtual_facts(self):
-        self.facts['virtualization_type'] = ''
-        self.facts['virtualization_role'] = ''
+        sysctl_path = self.module.get_bin_path('sysctl', required=True)
```
This fails the whole fact gathering if missing, i would just add an `if sysctl_path:` instead

eest replied:
@bcoca: I tested what would happen if i removed the sysctl binary on a testbox, and noticed the setup-module will already fail, because there are some places where the tool path is hardcoded and this leads to the module erroring out. This means my required=True addition does not make anything worse than it already is.

If the real solution is to also replace other instances of hardcoded sysctl paths (which would have to be a separate PR imo), then my question is what the desired behaviour is when the tool is not found: setting the effected fact variables to the empty string, or not setting them at all.

bcoca replied:
It's on my list to fix those, not asking you to do that, just this case so i have less work later (im selfish like that!).

I would say an empty string would be less error prone for uses so they don't have to check if the variable is defined.

eest committed:
Do not fail if a sysctl binary is not found.

Just set empty fact values instead.
Requested by @bcoca.

eest commented:
@bcoca: How about now? :)
